src/gui/guiChatConsole.cpp

```diff
@@ -710,35 +725,45 @@ void GUIChatConsole::setVisible(bool visible)
 	}
 }
 
-bool GUIChatConsole::weblinkClick(s32 col, s32 row)
-{
-	// Prevent accidental rapid clicking
-	static u64 s_oldtime = 0;
-	u64 newtime = porting::getTimeMs();
-
-	// 0.6 seconds should suffice
-	if (newtime - s_oldtime < 600)
-		return false;
-	s_oldtime = newtime;
-
+std::string GUIChatConsole::getLinkAt(s32 col, s32 row){
 	const std::vector<ChatFormattedFragment> &
-			frags = m_chat_backend->getConsoleBuffer().getFormattedLine(row).fragments;
+		frags = m_chat_backend->getConsoleBuffer().getFormattedLine(row).fragments;
 	std::string weblink = ""; // from frag meta
 
 	// Identify targetted fragment, if exists
 	int indx = frags.size() - 1;
 	if (indx < 0) {
 		// Invalid row, frags is empty
-		return false;
+		return "";
 	}
 	// Scan from right to left, offset by 1 font space because left margin
 	while (indx > -1 && (u32)col < frags[indx].column + 1) {
 		--indx;
 	}
 	if (indx > -1) {
 		weblink = frags[indx].weblink;
+		return weblink;
 		// Note if(indx < 0) then a frag somehow had a corrupt column field
 	}
+	return "";
+}
+
+bool GUIChatConsole::weblinkClickOpen(s32 col, s32 row)
```

Comment:
Please note that this PR does NOT add this code, all it did to it was changing it's name (which should be fine as it's private), and separate a part of it, as the feature added requires that as well. That somehow confused the git diff